src/app/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, OnInit } from '@angular/core';
+import { Http, Response } from '@angular/http';
+
+import { UserAuthenticationService } from '../api/UserAuthentication_service/UserAuthentication.service';
+import { WeatherAPIService } from '../api/WeatherAPI_service/WeatherAPI.service';
+import { ChangeScaleService } from '../api/ChangeScale_service/ChangeScale.service';
+import { Weather } from '../api/Classes/weather';
+import { User } from '../api/Classes/user';
+
+@Component({
+  selector: 'app-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.scss']
+})
+export class DashboardComponent implements OnInit {
+
+  currentUser: User;
+  email = 'anonymous@gmail.com';
+  history: string[] = this.login.getCurrentUser().history;
+  currentWeather: Weather;
+  temperature: any;
+
+  constructor(private login: UserAuthenticationService,
+              private dashboard: WeatherAPIService,
+              private header: ChangeScaleService,
+              private http: Http) {}
+
+  ngOnInit() {
+    this.currentUser = this.login.getCurrentUser();
+    this.history = this.currentUser.history;
+    this.email = this.login.getCurrentUser().email;
+  }
+
+  weather() {
+    return this.currentWeather == null ? false : true;
```

Comment:
always use !== or ===

Reply:
use Ecma Util function for this here to safeguard you from possible mistakes of using object in predicate function. 
```
weather() { //or whatever name should be here
   return  !isNullOrUndefined(this.currentWeather);
}
```